### DISCUSSION TITLE
How to create a Quarto website into single HTML file (and not multiple HTML files)

### DISCUSSION THREAD

**Discussion by Christophegauthierdavies** [quarto-dev/quarto-cli#6473] [Q&A]
### Description

I'm using Quarto with Rstudio to create a website. However the website is split into multiple HTML files related to each page. I would like the website to hold onto a single HTML file. The idea is to share this website with my colleague only (on our own server).

I've seen the _embed-resources_ function but it's not what I'm looking for.

Thank you.

**Comment by mcanouil** [comment]
Website are by definition multi HTML files.
This is not a Quarto limitation.

If you want a signe HTML document, the only way is to use `embed-resources: true` and use `format: html` but not as part of a book or website project.

**Reply by dmenne** [reply]
I am lost. I am trying to create a single report html (as I can for pdf) from multiple files. What is the equivalent of rmd:

```
bookdown::html_document2:
  keep_md: no
  toc_float: false
  delete_merged_file: true

book_filename: "MyBook"
new_session: no
clean: ["index.md"]
rmd_files: [
   "index.Rmd"
   ,"Exclusion.Rmd"

**Reply by mcanouil** [reply]
A book is not a report, even in the rmarkdown ecosytem.

It is very unclear to me what it is you want and from what.
Note that I believe, following the guides in the documentation can help you figure this out: <https://quarto.org/docs/guide/>.
For instance and because you seem to want to have a book: <https://quarto.org/docs/books/>.

**Reply by dmenne** [reply]
Sorry, I had forgotten why we stopped using Quarto already a year ago #3288

>> For instance and because you seem to want to have a book: https://quarto.org/docs/books/.
No, that will produce multiple files, and the solution in #3288 is ugly.

**Reply by mcanouil** [reply]
Indeed but you did not follow up on this, so we did not either unfortunately.

**Reply by cderv** [reply]
> What is the equivalent of rmd `html_document2`

There is currently no equivalent. **bookdown** has a special way of working by merging all the files before rendering to Pandoc , and then splitting it the resulting HTML to create a book. 

Quarto does not do any merging, and each files is rendered on its own, and then each HTML file is wrapped up in book-like envelop

I believe your question has already been discussed in 
* https://github.com/quarto-dev/quarto-cli/discussions/3267

And also for reference, here are related discussions (where you were probably involved already) 

* https://github.com/quarto-dev/quarto-cli/discussions/431
* https://github.com/quarto-dev/quarto-cli/discussions/1045
* https://github.com/quarto-dev/quarto-cli/discussions/3577
* https://github.com/quarto-dev/quarto-cli/discussions/2666

So currently not possible, maybe Quarto evolution in how we track and schedule rendering of files will allow this in the future;

**Comment by erblast** [comment]
check out 

https://github.com/gildas-lormeau/SingleFile-MV3/

its a browser extension that lets you download any webpage into a single file
